tools/ubuntu-setup/all.sh

```diff
@@ -1,23 +1,92 @@
+ERRORS=0
+
 echo "*** installing basics"
-source ./misc.sh
+source ./misc.sh &
+pid=$!
+wait $pid
+status=$?
+echo "$pid finished with status $status"
+if [ $status -ne 0 ]
+then
+    let ERRORS=ERRORS+1
+fi
```

Comment:
the above can be factored into a bash function so that it's not repeated per step.